### PR TITLE
Add a Slack posting resource

### PR DESCRIPTION
**Pull request by natekupp**

### CODE REVIEW

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class SlackConnection:
+    def __init__(self, token):
+        self.token = token
+        self.sc = SlackClient(self.token)
+
+    def post(
+        self,
+        channel='#noise',
+        username='dagsterbot',
+        text='https://www.youtube.com/watch?v=dQw4w9WgXcQ',
```
Comment (mgasner) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py]: 😭

**Comment by mgasner** [comment]
The official Slack client uses responses.RequestsMock to test this stuff: https://github.com/slackapi/python-slackclient/blob/master/tests/test_slackclient.py#L172

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from dagster import seven
+from dagster import Dict, Field, ResourceDefinition, String
+
+
+class SlackConnection:
+    def __init__(self, token):
+        self.token = token
+        self.sc = SlackClient(self.token)
+
+    def post(
```
Comment (schrockn) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py]: i think our naming convention for methods should directly mirror the namespace of their method calls. 

connection.chat.post_message

may seem natural

And I think a 

connection.api_call()

that does a direct pass through to the slack client make sense.

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            'icon_url': icon_url,
+            'attachments': seven.json.dumps(attachments),
+        }
+        self.sc.api_call(method, **api_params)
+
+
+def define_slack_resource():
+    return ResourceDefinition(
+        resource_fn=lambda init_context: SlackConnection(init_context.resource_config['token']),
+        config_field=Field(Dict({'token': Field(String)})),
```
Comment (schrockn) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py]: is there anything useful we could say about the token in the description?

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            'channel': channel,
+            'username': username,
+            'text': text,
+            'icon_url': icon_url,
+            'attachments': seven.json.dumps(attachments),
+        }
+        self.sc.api_call(method, **api_params)
+
+
+def define_slack_resource():
```
Comment (schrockn) [inline python_modules/libraries/dagster-slack/dagster_slack/resources.py]: i think we should encourage the use of @resource for simple resources like this

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py (hunk 2 above)
Comment (natekupp) [reply python_modules/libraries/dagster-slack/dagster_slack/resources.py]: yep, agree - updated

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py (hunk 4 above)
Comment (natekupp) [reply python_modules/libraries/dagster-slack/dagster_slack/resources.py]: fixed

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py (hunk 1 above)
Comment (natekupp) [reply python_modules/libraries/dagster-slack/dagster_slack/resources.py]: 🎉

File: python_modules/libraries/dagster-slack/dagster_slack/resources.py (hunk 3 above)
Comment (natekupp) [reply python_modules/libraries/dagster-slack/dagster_slack/resources.py]: yeah, added description w/ link to Slack API docs

**Comment by natekupp** [comment]
@mgasner great find w/ `responses`; I added a fairly basic test. lmk if you think we should have any others in this first PR

**Pull request merged**